src/emitterstate.cpp

```diff
@@ -124,10 +124,13 @@ void EmitterState::StartedGroup(GroupType::value type) {
   const int lastGroupIndent = (m_groups.empty() ? 0 : m_groups.top().indent);
   m_curIndent += lastGroupIndent;
 
-  std::auto_ptr<Group> pGroup(new Group(type));
+  std::unique_ptr<Group> pGroup(new Group(type));
```

Comment:
This bit all looks correct, but I think it can be all done easier with proper move constructors on the various settings types instead of using unique_ptr throughout. Can you put a TODO so that we can follow up later?

That is, write:

// TODO: Some text...